docs/ztp-precaching.md

```diff
@@ -116,7 +116,7 @@ spec:
next%20one%0A%20%20%20%20%20%20max_bg%3D10%0A%20%20%20%20%20%20pids%3D%28%29%0A%20%20%20%20fi%0A%20%20done%20%3C%20%24%7BBINARY_FOLDER%7D%2F%24%7BOCP_RELEASE_LIST%7D%0A%7D%0A%0Aretry_images%28%29%20%7B%0A%20%20echo%20%22%5BRETRYING%5D%22%0A%20%20echo%20%22%22%0A%20%20local%20rv%3D0%0A%20%20for%20failed_copy%20in%20%24%7Bfailed_copies%5B%40%5D%7D%3B%20do%0A%20%20%20%20%20%20echo%20%22%5BRETRY%5D%20Retrying%20failed%20image%20pull%3A%20%24%7Bfailed_copy%7D%22%0A%20%20%20%20%20%20tar%3D%24%28echo%20%22%24failed_copy%22%20%7C%20%20rev%20%7C%20cut%20-d%20%22%2F%22%20-f1%20%7C%20rev%20%7C%20tr%20%22%3A%22%20%22_%22%29%0A%20%20%20%20%20%20tar%20--use-compress-program%3Dpigz%20-xf%20%24%7Btar%7D.tgz%0A%20%20%20%20%20%20skopeo%20copy%20--retry-times%2010%20dir%3A%2F%2F%24%28pwd%29%2F%24%7Btar%7D%20containers-storage%3A%24%7Bfailed_copy%7D%20-q%0A%20%20%20%20%20%20if%20%5B%5B%20%24%3F%20%3D%3D%200%20%5D%5D%3B%20then%20%20%0A%20%20%20%20%20%20%20%20rm%20-rf%20%24%7Btar%7D%0A%20%20%20%20%20%20else%0A%20%20%20%20%20%20%20echo%20%22%5BERROR%5D%20Limit%20number%20of%20retries%20reached.%20The%20image%20could%20not%20be%20pulled%3A%20%24%7Bfailed_copy%7D%22%0A%20%20%20%20%20%20%20rv%3D1%0A%20%20%20%20%20%20fi%0A%20%20done%0A%20%20echo%20%22%5BINFO%5D%20Image%20load%20done%22%0A%20%20return%20%24rv%0A%7D%0A%0Aif%20%5B%5B%20%22%24%7BBASH_SOURCE%5B0%5D%7D%22%20%3D%20%22%24%7B0%7D%22%20%5D%5D%3B%20then%0A%20%20failed_copies%3D%28%29%20%23%20Array%20that%20will%20include%20all%20the%20images%20that%20failed%20to%20be%20pulled%0A%20%20load_images%0A%20%20retry_images%20%23%20Return%201%20if%20max.retries%20reached%0A%20%20if%20%5B%5B%20%24%3F%20-ne%200%20%5D%5D%3B%20then%0A%20%20%20%20echo%20%22%5BFAIL%5D%20One%20or%20more%20images%20were%20not%20precached%20successfully%22%0A%20%20%20%20exit%201%0A%20%20else%0A%20%20%20%20echo%20%22%5BSUCCESS%5D%20All%20images%20were%20precached%22%0A%20%20%20%20exit%200%0A%20%20fi%0Afi"}},{"overwrite":true,"path":"/usr/local/bin/agent-fix-bz1964591","mode":755,"user":{"name":"root"},"contents":{"source":"data:,%23%21%2Fusr%2Fbin%2Fsh%0A%0A%23%20This%20script%20is%20a%20workaround%20for%20bugzilla%201964591%20where%20symlinks%20inside%20%2Fvar%2Flib%2Fcontainers%2F%20get%0A%23%20corrupted%20under%20some%20circumstances.%0A%23%0A%23%20In%20order%20to%20let%20agent.service%20start%20correctly%20we%20are%20checking%20here%20whether%20the%20requested%0A%23%20container%20image%20exists%20and%20in%20case%20%22podman%20images%22%20returns%20an%20error%20we%20try%20removing%20the%20faulty%0A%23%20image.%0A%23%0A%23%20In%20such%20a%20scenario%20agent.service%20will%20detect%20the%20image%20is%20not%20present%20and%20pull%20it%20again.%20In%20case%0A%23%20the%20image%20is%20present%20and%20can%20be%20detected%20correctly%2C%20no%20any%20action%20is%20required.%0A%0AIMAGE%3D%24%28echo%20%241%20%7C%20sed%20%27s%2F%5B%40%3A%5D.%2A%2F%2F%27%29%0Apodman%20images%20%7C%20grep%20%24IMAGE%20%7C%7C%20podman%20rmi%20--force%20%241%20%7C%7C%20true"}}]}}'
```

Comment:
Should we use base64 encoding for the content in these large script ignitionConfigOverrides?